cabal-install/Distribution/Client/CmdRepl.hs

```diff
@@ -272,6 +272,9 @@ replAction (configFlags, configExFlags, installFlags, haddockFlags, testFlags, r
           replFlags' = case originalComponent of 
             Just oci -> generateReplFlags includeTransitive elaboratedPlan' oci
             Nothing  -> []
+          replFlags'' = case replType of
+            GlobalRepl scriptPath -> ("-ghci-script" ++ scriptPath) : replFlags'
```

Comment:
this is funny. it seems it doesn't actually require a space in between. However:

```
% ghci-7.0.4 -ghci-script foo
GHCi, version 7.0.4: http://www.haskell.org/ghc/  :? for help
ghc: unrecognised flags: -ghci-script
Usage: For basic information, try the `--help' option.
```

```
% ghci-7.4.2 -ghci-script foo
GHCi, version 7.4.2: http://www.haskell.org/ghc/  :? for help
ghc: unrecognised flags: -ghci-script
Usage: For basic information, try the `--help' option.
```

```
% ghci-7.6.3 -ghci-script foo
GHCi, version 7.6.3: http://www.haskell.org/ghc/  :? for help
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
"foo"
```

So for pre-GHC-7.6 we need to do something else, or at least don't pass `-ghci-script`.

Reply:
ref https://downloads.haskell.org/~ghc/7.6.1/docs/html/users_guide/release-7-6-1.html

> GHCi now supports loading additional `.ghci` files via the `-ghci-script` flag

Reply:
And to clarify, I personally will be ok if out-of-project `cabal repl` will be somewhere not in CWD (due to limitations of old GHC) , but it shouldn't fail to load.

Reply:
Added a check, now it just puts you in the wrong directory rather than failing on dinosauric GHCs. 🙂